spec/bolt/executor_spec.rb

```diff
@@ -310,6 +310,25 @@ def mock_node_results(_run_as = nil)
         expect(result.error_hash['kind']).to eq('puppetlabs.tasks/exception-error')
       end
     end
+
+    context 'with batched execution with more than one target' do
+      let(:pcp) { executor.transport('pcp') }
+      let(:target_1) { Bolt::Target.new('pcp://node1').update_conf(Bolt::Config.default.transport_conf) }
+      let(:target_2) { Bolt::Target.new('pcp://node2').update_conf(Bolt::Config.default.transport_conf) }
+      let(:targets) { [target_1, target_2] }
+      
+      it 'partitions failures and successes by batch' do
+        allow(pcp).to receive(:batch_connected?).with(targets).and_return(false)
+        allow(pcp).to receive(:batch_connected?).with([target_1]).and_return(false)
+        allow(pcp).to receive(:batch_connected?).with([target_2]).and_return(true)
+        expect(executor).to receive(:wait_now).and_return(Time.now - 1, Time.now, Time.now + 1)
```

Comment:
OOps, meant to delete these two expectations. Will do once I verify CI is green.